### PR TITLE
feat(AUTH-CPC-742): create account

### PR DESCRIPTION
**Pull request by RichardDanon**
JIRA: https://champlainsaintlambert.atlassian.net/browse/CPC-742
## Context:
This ticket is about creating an account to log into the Pet Clinic app.
Its necessity is self-explanatory since an account is required to use the application.

## Changes
What are the various changes and what other modules do those changes affect?

1. Email is not linked to any account
2. Password should pass strength checks
3. The user receives an email and accepts the email verification
4. Implement email verification and any other field verification needed. (@Email for example)
5. Sending an email to the user to confirm the email and this should set the user to verified
6. linking the default customer role to new users
7. Create an Owner role
8. Application Tested

 ## Before and After UI (Required for UI-impacting PRs)

image
image

## Dev notes (Optional)
- Specific technical changes that should be noted
## Linked pull requests (Optional)
- pull request link

### CODE REVIEW

File: auth-service/src/main/java/com/auth/authservice/Util/Configuration/Security/PasswordValidation.java [hunk 1]
```diff
@@ -21,11 +22,18 @@ public void initialize(PasswordStrengthCheck constraintAnnotation) {
     @Override
     public boolean isValid(String value, ConstraintValidatorContext context) {
 
-        String pattern = "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+=])(?=\\S+$).{8,}$";
+        if (value == null) {
+            return false;
+        }
+        log.info("Where is my password");
 
+        String pattern = "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+=*])(?=\\S+$).{8,}$";
```
Comment (DylanBrass) [inline auth-service/src/main/java/com/auth/authservice/Util/Configuration/Security/PasswordValidation.java:30]: Again, not in your sprint but a great initiative !

File: auth-service/src/main/java/com/auth/authservice/Util/Configuration/Security/SecurityConfig.java [hunk 2]
```diff
@@ -70,6 +71,8 @@ public AuthenticationManager authenticationManager() {
     public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
         http.authorizeHttpRequests(authorizeRequests -> authorizeRequests
                 .requestMatchers(AntPathRequestMatcher.antMatcher(HttpMethod.GET,"/users/verification/*")).permitAll()
+                .requestMatchers(AntPathRequestMatcher.antMatcher(HttpMethod.POST, "/users/verification/**")).permitAll()
+                // should skip filters
```
Comment (DylanBrass) [inline auth-service/src/main/java/com/auth/authservice/Util/Configuration/Security/SecurityConfig.java:75]: Try to make the endpoints more specific.

File: auth-service/src/main/java/com/auth/authservice/datalayer/user/User.java [hunk 3]
```diff
@@ -56,6 +57,7 @@ public class User implements UserDetails {
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private long id;
 
+    //private String userId;
```
Comment (DylanBrass) [inline auth-service/src/main/java/com/auth/authservice/datalayer/user/User.java:60]: To be done in another story.

File: auth-service/src/main/java/com/auth/authservice/security/JwtTokenFilter.java [hunk 4]
```diff
@@ -77,7 +77,7 @@ protected void doFilterInternal(HttpServletRequest request,
         if (sessionCookie == null) {
             log.info("No token found");
             response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-            chain.doFilter(request, response);
+            resolver.resolveException(request, response, null, new InvalidBearerTokenException("No Token Found"));
             return;
```
Comment (DylanBrass) [inline auth-service/src/main/java/com/auth/authservice/security/JwtTokenFilter.java:81]: We did this together.

**Review by DylanBrass — DISMISSED** [review]

**Review by DylanBrass — DISMISSED** [review]

File: auth-service/src/main/java/com/auth/authservice/Util/GlobalControllerExceptionHandlerConfig.java [hunk 5]
```diff
@@ -26,6 +30,26 @@ public HTTPErrorMessage incorrectPasswordException(IncorrectPasswordException ex
         return new HTTPErrorMessage(UNAUTHORIZED.value(), ex.getMessage());
     }
 
+//    @ExceptionHandler(value = MethodArgumentNotValidException.class)
```
Comment (DenisaNicoletaH) [inline auth-service/src/main/java/com/auth/authservice/Util/GlobalControllerExceptionHandlerConfig.java:33]: If this is useless, you should remove this block of commented code

**Review by DenisaNicoletaH — APPROVED** [review]
After seeing the changes that was made, it looks better! The requirements and the naming convention are followed.

**Review by DylanBrass — DISMISSED** [review]

**Review by DenisaNicoletaH — APPROVED** [review]

File: auth-service/src/main/java/com/auth/authservice/security/JwtTokenFilter.java (hunk 4 above)
Comment (DylanBrass) [reply auth-service/src/main/java/com/auth/authservice/security/JwtTokenFilter.java:81]: Bu it should be chain.doFilter

**Review by DylanBrass — APPROVED** [review]

**Review by DenisaNicoletaH — APPROVED** [review]

**Pull request merged**